posthog/models/property.py

```diff
@@ -37,47 +53,114 @@
     "is_date_before",
 ]
 
+OperatorInterval = Literal["day", "week", "month", "year"]
 GroupTypeName = str
 PropertyIdentifier = Tuple[PropertyName, PropertyType, Optional[GroupTypeIndex]]
 
 NEGATED_OPERATORS = ["is_not", "not_icontains", "not_regex", "is_not_set"]
 CLICKHOUSE_ONLY_PROPERTY_TYPES = ["static-cohort", "precalculated-cohort"]
 
+VALIDATE_PROP_TYPES = {
+    "event": ["key", "value"],
+    "person": ["key", "value"],
+    "cohort": ["key", "value"],
+    "element": ["key", "value"],
+    "static-cohort": ["key", "value"],
+    "precalculated-cohort": ["key", "value"],
+    "group": ["key", "value", "group_type_index"],
+    "recording": ["key", "value"],
+    "performed_event": ["event", "event_type", "time_value", "time_interval"],
+    "performed_event_multiple": ["event", "event_type", "time_value", "time_interval", "operator", "operator_value"],
+}
+
 
 class Property:
```

Comment:
Open to other ideas here but I wanted to avoid splitting out objects unnecessarily